src/lib/homeobject_impl.hpp

```diff
@@ -21,15 +16,32 @@ namespace homeobject {
 inline bool operator<(ShardInfo const& lhs, ShardInfo const& rhs) { return lhs.id < rhs.id; }
 inline bool operator==(ShardInfo const& lhs, ShardInfo const& rhs) { return lhs.id == rhs.id; }
 
+struct ShardInfoExt {
+    uint16_t chunk_id;
+};
+
+struct Shard {
+    explicit Shard(ShardInfo info) : info(std::move(info)) {}
+    ShardInfo info;
+    ShardInfoExt ext_info;
+};
+
+using ShardPtr = std::shared_ptr< Shard >;
+using ShardPtrList = std::list< ShardPtr >;
+
+struct PG {
+    explicit PG(PGInfo info) : pg_info(std::move(info)) {}
+    PGInfo pg_info;
+    uint64_t shard_sequence_num{0};
+    ShardPtrList shards;
```

Comment:
We don't need `Shard` to be in a shared ptr; this can just be `ShardList` and we hold iterators in the `_shard_map` since the list will never be mutated. We can tombstone deleted shards etc. Removes an atomic counter that's not serving a purpose.

Reply:
ok, will do that.